feat: add i18n error message support for TimePicker

vursen opened the pull request:
## Description

The PR introduces TimePickerI18n with methods for setting error messages and updates the TimePicker's validation logic to show these error messages when validation fails.

> [!NOTE]
> It's still possible to use the `setErrorMessage` method to configure a single static error message. This error message will be displayed for all constraints and will take priority over any i18n error messages that are also set. However, when more than one error message is needed, i18n should be used or manual validation mode should be enabled.

Part of #4618 

## Type of change

- [x] Feature

DiegoCardoso commented on vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java:
```diff
@@ -680,4 +693,163 @@ public static Stream<Locale> getSupportedAvailableLocales() {
+                "The i18n properties object should not be null");
+    }
+
+    private String getBadInputErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(TimePickerI18n::getBadInputErrorMessage).orElse("");
+    }
+
+    private String getRequiredErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(TimePickerI18n::getRequiredErrorMessage).orElse("");
+    }
+
+    private String getMinErrorMessage() {
+        return Optional.ofNullable(i18n).map(TimePickerI18n::getMinErrorMessage)
+                .orElse("");
+    }
+
+    private String getMaxErrorMessage() {
+        return Optional.ofNullable(i18n).map(TimePickerI18n::getMaxErrorMessage)
+                .orElse("");
+    }
```
Nit: as this pattern seems to be something used multiple times not only in this file, but in other components too, would it make sense to extract it as helper function like:

```java
private <T> String getMessageOrEmptyString(T object, Function<T, String> getter) {
    return Optional.ofNullable(object).map(getter).orElse("");
}
```

It would be called as `getMessageOrEmptyString(i18n, TimePickerI18n::getMinErrorMessage)`, for instance.

vursen replied:
It's a good point. How about this?

```java
private String getI18nErrorMessage(
        Function<DateTimePickerI18n, String> getter) {
    return Optional.ofNullable(i18n).map(getter).orElse("");
}
```

DiegoCardoso replied:
For using it just inside of this class, then the generic is not needed 👍🏻

vursen replied:
Updated.

vursen commented:
During our internal discussion, we discovered a way to keep error messages that are set with `setErrorMessage`. They now take priority over i18n error messages, which allows developers to continue using the old familiar method, setting a single error message for all constraints. I've updated the PR and also removed the breaking change mark.
